assets/js/blocks/checkout/inner-blocks/checkout-collection-method/frontend.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { withFilteredAttributes } from '@woocommerce/shared-hocs';
+import { FormStep } from '@woocommerce/base-components/cart-checkout';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
+/**
+ * Internal dependencies
+ */
+import Block from './block';
+import attributes from './attributes';
+
+const FrontendBlock = ( {
+	title,
+	description,
+	showStepNumber,
+	children,
+	className,
+	showPrice,
+	showIcon,
+}: {
+	title: string;
+	description: string;
+	showStepNumber: boolean;
+	children: JSX.Element;
+	className?: string;
+	showPrice: boolean;
+	showIcon: boolean;
+} ) => {
+	const { checkoutIsProcessing, prefersCollection } = useSelect(
+		( select ) => {
+			const checkoutStore = select( CHECKOUT_STORE_KEY );
+			return {
+				checkoutIsProcessing: checkoutStore.isProcessing(),
+				prefersCollection: checkoutStore.prefersCollection(),
+			};
+		}
+	);
+	const { setPrefersCollection } = useDispatch( CHECKOUT_STORE_KEY );
+
+	const onChange = ( method: string ) => {
+		if ( method === 'pickup' ) {
+			setPrefersCollection( true );
+		} else {
+			setPrefersCollection( false );
+		}
+	};
+	return (
+		<FormStep
+			id="collection-method"
+			disabled={ checkoutIsProcessing }
+			className={ classnames(
+				'wc-block-checkout__collection-method',
+				className
+			) }
+			title={ title }
+			description={ description }
+			showStepNumber={ showStepNumber }
+		>
+			<Block
+				checked={ prefersCollection ? 'pickup' : 'shipping' }
```

Comment:
`pickup` and `collection` are pretty much the same thing. We should again agree on the term to use here.